File: 2020/3/lib/parser.rb
# frozen_string_literal: true

class Parser
  def self.parse(input = nil)
    input ||= File.read(File.join(File.dirname(__FILE__), "../input.txt"))

    input
      .strip
      .split("\n")
      .map(&:chars)
      .each_with_object({}).with_index do |(line, acc), y|
        line.each_with_index do |e, x|
          acc[[x, y]] = e
        end
      end
  end
end

File: 2016/02/lib/aocday/parser.ex
defmodule AOCDay.Parser do
  def parse("") do
    {:ok, binary} = File.read("./input.txt")
    _parse(binary)
  end

  def parse(binary) do
    _parse(binary)
  end

  defp _parse(binary) do
    binary
    |> String.trim
    |> String.split("\n")
    |> Enum.map(fn(line) -> String.trim(line) |> String.graphemes end)
  end
end
